include/eve/detail/shuffle_v2/shuffle_v2_helpers.hpp

```diff
@@ -21,11 +21,23 @@ constexpr no_matching_shuffle_t no_matching_shuffle;
 template<typename T>
 concept matched_shuffle = !std::same_as<no_matching_shuffle_t, std::remove_cvref_t<T>>;
 
+template <typename T>
+concept shuffle_user_error = requires (T a) {
+  typename T::is_shuffle_user_error;
+};
+
 EVE_FORCEINLINE auto
 invoke_shuffle(auto sh, auto p, auto g, auto... xs)
 {
   if constexpr( requires { sh(p, g, xs...); } ) { return sh(p, g, xs...); }
   else return no_matching_shuffle;
 }
 
+EVE_FORCEINLINE auto
+invoke_shuffle_multilevel(auto sh, auto p, auto g, auto... xs)
```

Comment:
I group all shuffles that are N instructions in one function after level 3, because we recurse.
But this leads to one function can have multiple levels.  At this point this is flimsy, we may revisit levels later.